src/main/java/org/everit/jira/timetracker/plugin/JiraTimetrackerAnalytics.java

```diff
@@ -0,0 +1,131 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.everit.jira.timetracker.plugin;
+
+import java.io.UnsupportedEncodingException;
+import java.security.NoSuchAlgorithmException;
+
+import javax.servlet.http.HttpSession;
+
+import org.apache.log4j.Logger;
+
+import com.atlassian.jira.component.ComponentAccessor;
+
+/**
+ * The Jira Timetracker plugin analytics class.
+ */
+public final class JiraTimetrackerAnalytics {
+
+  /**
+   * The JiraTimetrackerAnalytics logger.
+   */
+  private static Logger log = Logger.getLogger(JiraTimetrackerAnalytics.class);
+
+  private static final String ERROR_BASE_URL_HASH = "errorBaseUrlHash";
+
+  private static final String ERROR_USER_ID_HASH = "errorUserIdHash";
+
+  /**
+   * Get the base URL.
+   *
+   * @return The base URL.
+   */
+  public static String getBaseUrl() {
+    String baseUrl;
+    try {
+      baseUrl = Hash.encryptString(ComponentAccessor
+          .getApplicationProperties()
+          .getString("jira.baseurl"));
+      return baseUrl;
+    } catch (NoSuchAlgorithmException | UnsupportedEncodingException e) {
+      log.error("Error when try to hash the base URL.", e);
+      return ERROR_BASE_URL_HASH;
+
+    }
+
+  }
+
+  /**
+   * Get the version of the plugin.
+   *
+   * @return The version.
+   */
+  public static String getPluginVersion() {
+    final String pluginVersion = ComponentAccessor.getPluginAccessor()
+        .getPlugin("org.everit.jira.timetracker.plugin")
+        .getPluginInformation().getVersion();
+    return pluginVersion;
+  }
+
+  /**
+   * Get the user ID.
+   *
+   * @return The user.
+   */
+  public static String getUserId() {
+    String userId;
+    try {
+      userId = Hash.encryptString(ComponentAccessor
+          .getJiraAuthenticationContext().getUser().getKey());
+    } catch (NoSuchAlgorithmException | UnsupportedEncodingException e) {
+      log.error("Error when try to hash the user ID.", e);
+      return ERROR_USER_ID_HASH;
+    }
+    return userId;
+  }
+
+  /**
+   * Set the hashed base URL.
+   *
+   * @param session
+   *          Session.
+   * @return The hashed base URL.
+   */
+  public static String setUserSessionBaseUrl(final HttpSession session) {
+    String resultBaseUrl;
+    final String hashedBaseUrl = (String) session.getAttribute("BASE_URL");
+    if (hashedBaseUrl == null) {
+      resultBaseUrl = JiraTimetrackerAnalytics.getBaseUrl();
+      session.setAttribute("BASE_URL", resultBaseUrl);
+    } else {
+      resultBaseUrl = hashedBaseUrl;
+    }
+    return resultBaseUrl;
+  }
+
+  /**
+   * Set the hashed user ID.
+   *
+   * @param session
+   *          Session
+   * @return The hashed user ID.
+   */
+  public static String setUserSessionUserId(final HttpSession session) {
+    String resultUserId;
+    final String hashedUserId = (String) session.getAttribute("USER_ID");
```

Comment:
Extract the "USER_ID" to a constant and make it more unique. The HttpSession can be accessed "everywhere" in a servlet container. For example: "org.everit.jira.timetracker.plugin.user.id.hash".